### ISSUE TITLE
Using the Resource Name as a key value

### ISSUE THREAD

**Issue by AshutoshDahale** [aws-cloudformation/cloudformation-guard#281]
**Describe the issue**
Hi All, 
I am just wondering if I we can use the Resource Name in the Cloud formation as a value say type of string so that that can be used to compare with some of the Referenced values in some other Resource in the template.
Please let me know if we can do that, if not this can be a good addition to the tool.

**Any examples**
For Example I have the Resource of type SecurityGroupIngress as follows: 

```yaml:
SecurityConfiguratorSecurityGroupToElasticSearch:
    Type: AWS::EC2::SecurityGroupIngress
    Properties:
      Description: !Sub "Allow ElasticSearch from ${Environment}-Elasticsearch security configuration Lambda"
      GroupId:
        Fn::ImportValue:
          !Sub "${Environment}-SecurityGroups-ElasticSearch"
      SourceSecurityGroupId: SecurityConfigurationSecurityGroup
      IpProtocol: tcp
      FromPort: 443
      ToPort: 443
```

In the above example I have refered the "SecurityConfigurationSecurityGroup" Resource, ultimatelly saying that this ingress rule is for that SecurityGroup Rule.
Now I need to validate if Every security group has the ingress rule in place for that I need to compair the reference "SourceSecurityGroupId"  with the Rules available in the Template.

I have tried the following rule:

```guard:
rule SecurityGroupIngress when Resources.*[ Type == 'AWS::EC2::SecurityGroupIngress' ] !empty {
   Resources.*[ Type == 'AWS::EC2::SecurityGroupIngress' ].Properties {
      Resources.*[ Type == 'AWS::EC2::SecurityGroupIngress' ].Properties.SourceSecurityGroupId[*] IN Resources.* <<its rule ingress failing>>
}}
```

**Comment by akshayrane** [comment]
@AshutoshDahale Thanks for reaching out to us with this issue. We can see that you are trying to look up ingress rules' `SourceSecurityGroupId` property but it's not very clear from the description what are we trying to compare it with.  

Can you re-word the description of the rule in plain English so we could understand better?

Are you trying to do the following?

> Lookup `SourceSecurityGroupId` for all resources of type `AWS::EC2::SecurityGroupIngress` and then check whether these IDs are part of a list of hard-coded IDs

**Comment by akshayrane** [comment]
@AshutoshDahale  - Can you confirm if our understanding of the problem is correct before we move forward? Or if not, help us with your understanding?

If we do not get a response, we may mark this issue as closed.

**Comment by razcloud** [comment]
Closing.  Please let us know if you need additional assistance.
